### PR TITLE
AAAA record support for digital_ocean_domain

### LINKED ISSUE #226: AAAA record support for digital_ocean_domain
##### SUMMARY
The digital_ocean_domain module does not support creating a `AAAA` record which is an essential feature.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
digital_ocean_domain

### PR DESCRIPTION
**Pull request by mamercad**
##### SUMMARY

Fixes #226; I'm going to tackle individual DNS records with a new module, `digital_ocean_domain_record` (it'll be cleaner this way).

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

* digital_ocean_domain

### CODE REVIEW

File: plugins/modules/digital_ocean_domain.py [hunk 1]
```diff
@@ -155,7 +161,11 @@ def destroy_domain(self):
             return json
 
     def edit_domain_record(self, record):
-        params = {"name": "@", "data": self.module.params.get("ip")}
+        if self.module.params.get("ip"):
+            params = {"name": "@", "data": self.module.params.get("ip")}
+        if self.module.params.get("ip6"):
+            params = {"name": "@", "data": self.module.params.get("ip6")}
```
Comment (mpontillo) [inline plugins/modules/digital_ocean_domain.py:167]: The `params` here is being overwritten; does this mean that this module only supports setting one type of apex record at a time?

Comment (mamercad) [reply plugins/modules/digital_ocean_domain.py:167]: The [API docs](https://docs.digitalocean.com/reference/api/api-reference/#operation/create_domain_record) aren't overly clear, but yes, it looks like the `type` parameter can be *one of* `['A', 'AAAA', 'CNAME', ...]`.

Comment (mamercad) [reply plugins/modules/digital_ocean_domain.py:167]: The [API docs](https://docs.digitalocean.com/reference/api/api-reference/#operation/create_domain_record) aren't overly clear, but yes, it looks like the `type` parameter can be *one of* `['A', 'AAAA', 'CNAME', ...]`.

Comment (mpontillo) [reply plugins/modules/digital_ocean_domain.py:167]: So should we make this an `elif`, then, and document the idea that `ip` will be preferred over `ip6`? (as written, if both are supplied then only `ip6` will be updated)

Comment (mamercad) [reply plugins/modules/digital_ocean_domain.py:167]: I don't think it matters; this module uses `mutually_exclusive=[("ip", "ip6")],` -- the module will fail if both arguments are specified.

**Review by mpontillo — APPROVED** [review]

**Pull request merged**